modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts

```diff
@@ -361,6 +389,92 @@ export class LinkModalDialog
         return this.mediaOptionRadioFormItem;
     }
 
+    private createUrlFormItem(textId: string, textLabel: string): FormItem {
+        const getUrl = () => this.isUrl() ? this.link : '';
+
+        const urlFormItem: FormItem = this.createFormItemWithPostponedValue(textId, textLabel, getUrl);
+        const urlInput = <TextInput>urlFormItem.getInput();
+
+        urlInput.onRendered(() => {
```

Comment:
Try to avoid long method bodies. If you assign several event listeners and/or listeners have long handlers, it makes sense to extract this to a separate method.

For example,

```Typescript
private createUrlFormItem() {
  const urlInput = <TextInput>urlFormItem.getInput();
  this.initUrlInputHandlers(urlInput);
  ...
}

private initUrlInputHandlers(urlInput: TextInput) {
  urlInput.onRendered(() => {
  });
  urlInput.onValueChanged(() => {
  });
}
```